Projects/ChangeTier/Sources/Views/CompareTierScreen.swift

```diff
@@ -87,4 +77,94 @@ extension ChangeTierViewModel {
             })
         return tempPerils
     }
+
+    @MainActor
+    public func getProductVariantComparision() async throws -> ProductVariantComparison? {
+        withAnimation {
+            viewState = .loading
+        }
+        do {
+            var termsVersionsToCompare: [String] = []
+            tiers.forEach({ tier in
+                tier.quotes.forEach({ quote in
+                    if let termsVersion = quote.productVariant?.termsVersion,
+                        !termsVersionsToCompare.contains(termsVersion)
+                    {
+                        termsVersionsToCompare.append(termsVersion)
+                    }
+                })
+            })
+
+            let productVariantComparisionData = try await service.compareProductVariants(
+                termsVersion: termsVersionsToCompare
+            )
+
+            withAnimation {
+                viewState = .success
+            }
+            return productVariantComparisionData
+        } catch let error {
+            withAnimation {
+                self.viewState = .error(
+                    errorMessage: error.localizedDescription
+                )
+            }
+        }
+        return nil
+    }
+}
+
+struct CompareTierScreen: View {
+    @ObservedObject private var vm: CompareTierViewModel
+    @EnvironmentObject var changeTierNavigationVm: ChangeTierNavigationViewModel
+
+    init(
+        vm: CompareTierViewModel
+    ) {
+        self.vm = vm
+    }
+
+    var body: some View {
+        ProcessingStateView(
```

Comment:
We don't want to have this loader I have added 
    public func loading(_ state: Binding<ProcessingState>) -> some View {
        modifier(LoadingViewWithContentForProcessingState(state: state))
    }
just use:
```
succesView.loading($vm.viewState)
        .hErrorViewButtonConfig(
            .init(
                actionButton: .init(
                    buttonAction: {
                        Task {
                            try await vm.getProductVariantComparision()
                        }
                    }
                ),
                dismissButton:
                    .init(
                        buttonTitle: L10n.generalCloseButton,
                        buttonAction: {
                            changeTierNavigationVm.router.dismiss()
                        }
                    )
            )
        )

```